controller/internal/enforcer/envoyauthorizer/envoyproxy/auth_server.go

```
@@ -0,0 +1,466 @@
+func (s *AuthServer) egressCheck(ctx context.Context, checkRequest *ext_auth.CheckRequest) (*ext_auth.CheckResponse, error) {
+	// pctxRaw, err := s.puContexts.Get(s.puID)
+	// if err != nil {
+	// 	zap.L().Error("ext_authz ingress: failed to get PU context", zap.String("puID", s.puID), zap.Error(err))
+	// 	return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "failed to get PU context"), nil
+	// }
+	// pctx, ok := pctxRaw.(*pucontext.PUContext)
+	// if !ok {
+	// 	zap.L().Error("ext_authz ingress: PU context has the wrong type", zap.String("puID", s.puID), zap.String("puContextType", fmt.Sprintf("%T", pctxRaw)))
+	// 	return createDeniedCheckResponse(rpc.INTERNAL, envoy_type.StatusCode_InternalServerError, "PU context has the wrong type"), nil
+	// }
+	var sourceIP, destIP string
+	var source, dest *ext_auth.AttributeContext_Peer
+	var httpReq *ext_auth.AttributeContext_HttpRequest
+	var destPort, srcPort int
+	var urlStr, method string
+	attrs := checkRequest.GetAttributes()
+	if attrs != nil {
+		source = attrs.GetSource()
+		dest = attrs.GetDestination()
+
+		if source != nil {
+			if addr := source.GetAddress(); addr != nil {
+				if sockAddr := addr.GetSocketAddress(); sockAddr != nil {
+					sourceIP = sockAddr.GetAddress()
+					srcPort = int(sockAddr.GetPortValue())
+				}
+			}
+		}
+		if dest != nil {
+			if destAddr := dest.GetAddress(); destAddr != nil {
+				if destSockAddr := destAddr.GetSocketAddress(); destSockAddr != nil {
+					destIP = destSockAddr.GetAddress()
+					destPort = int(destSockAddr.GetPortValue())
+				}
+			}
+		}
+
+		if request := attrs.GetRequest(); request != nil {
+			httpReq = request.GetHttp()
+			urlStr = httpReq.GetPath()
+			method = httpReq.GetMethod()
+		}
+	}
+	// Create the new target URL based on the path parameter that we have from envoy.
+	URL, err := url.ParseRequestURI(urlStr)
+	if err != nil {
+		zap.L().Debug("ext_authz egress: Cannot parse the URI")
+		return nil, nil
```

should return the error?